File: elemental/src/dto/InterazioneDTO.ts
import {UserDTO} from "./userdto";
import {PostDTO} from "./postdto";

export class InterazioneDTO {
  id: number

  likes: number

  user: UserDTO

  post: PostDTO

  constructor(
  id: number,

  likes: number,

  user: UserDTO,

  post: PostDTO,
  ) {
    this.id = id
    this.likes = likes
    this.user = user
    this.post = post
  }
}

File: elemental-service/src/main/java/com/simonecampis/Elemental/dto/InterazioneDTO.java
package com.simonecampis.Elemental.dto;

import com.simonecampis.Elemental.model.Post;
import com.simonecampis.Elemental.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class InterazioneDTO {

    private Long id;

    private Long likes;

    private User user;

    private Post post;

}
